_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,236 @@
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Information and relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/WAI/WCAG21/Techniques/html/H43)
+- [Forming relationships between data cells and header cells](https://html.spec.whatwg.org/multipage/tables.html#header-and-data-cell-semantics)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Each `th` element has corresponding cells within the same `table`.
+
+```html
+<table>
+	<tr>
+		<th>Time</th>
+	</tr>
+	<tr>
+		<td>05:41</td>
+	</tr>
+</table>
+```
+
+#### Passed Example 2
+
+Each `th` element has corresponding cells within the same `table`. Here the cells span multiple columns.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th>Projects</th>
+      <th>Exams</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td colspan="2">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 3
+
+The `th` element has corresponding cells within the same `table`, by usage of `headers` attribute.
+
+```html
+<table>
+  <tr> 
+    <td headers="header1">
+      Projects
+    </td>
+    <th id="header1">
+      Projects
+    </th> 
+  </tr>
+</table>
+```
+
+#### Passed Example 4
+
+The `th` element has corresponding cells within the same `table`, by usage of `aria-labelledby` attribute.
+
+```html
+<table>
+	<tr>
+    <th id="hd2">Time</th>
+    <td aria-labelledby="hd2">05:44</td>
+	</tr>
+</table>
+```
+
+#### Passed example 5
+
+Each `th` element has corresponding cells within the same `table`, by usage of `scope` attribute.
+
+```html
+<table>
+  <caption>Hogwarts Houses stars</caption>
+  <tr>
+      <th scope="col">Name</th>
+      <th scope="col">House</th>
+  </tr>
+  <tr>
+      <th scope="row">Harry Potter</th>
+      <td>Gryffindor</td>
+  </tr>
+  <tr>
+      <th scope="row">Cedric Diggory</th>
+      <td>Hufflepuff</td>
+  </tr>
+</table>
+```
+
+#### Passed example 6
+
+Each `th` element has corresponding cells within the same `table`.
+
+```html
+<table>
+	<caption>Hogwarts Houses stars</caption>
+	<tr>
+			<th>Name</th>
+			<th>House</th>
+	</tr>
+	<tr>
+			<th>Cho Chang</th>
+			<td>Ravenclaw</td>
+	</tr>
+	<tr>
+			<th>Severus Snape</th>
+			<td>Slytherin</td>
+	</tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+`th` elements within the `table` do not have corresponding cells.
+
+```html
+<table>
+	<tr>
+		<th>Header 1</th>
+	</tr>
+	<tr>
+		<th>Header 2</th>
+	</tr>
+</table>
```

Comment:
@jkodu If I recall correctly, didn't we discuss restricting the applicability of this rule to tables that actually have data cells in order for this case to pass? What are your thoughts on this, @ShadowBB? Would you regard an empty table consisting only of headers to be an accessibility issue and, perhaps more importantly, a failure of 1.3.1?

Reply:
I would fail that on 1.3.1.

1.3.1 demands that relationships are communicated. If we imply the existence of relationships that are not actually there we devalue the information about correct relationships. I also fail non-table heading elements without associated content.

But I agree that it is a stretch because no actual wrong relationship has been communicated, only implied. (Communicating a relationship would need the involvement of at least 2 elements instead of the incorrect role on just a single one).

Reply:
Discussed with @kasperisager and based on above comment from @ShadowBB, makes sense to leave this as is and without any change.

Reply:
We discussed this in the face to face meeting and Wilco convinced me that my explanation above is incorrect.

We fail non-table-headings without content because it messes up the page structure(!) not it's relationships. And table headings don't communicate page structure.

So this example should become not applicable and we should change the applicability of this rule to reflect this.